src/groupederror.jl

```diff
@@ -202,14 +204,24 @@ Output is a `GroupedError` with error computed according to the keyword `compute
 It can be plotted using `plot(g::GroupedError)`
 Seriestype can be specified to be `:path`, `:scatter` or `:bar`
 """
-function groupapply(f::Function, df, args...;
-                    axis_type = :auto, compute_error = :none, group = [],
+function groupapply(f::Function, dfn, args...;
+                    axis_type = :auto, compute_error = :none, group = Array{Symbol,1}(0),
                     summarize = (get_symbol(compute_error) == :bootstrap) ? (mean, std) : (mean, sem),
                     compute_axis = :separate,
                     kwargs...)
+
     if !(axis_type in [:discrete, :continuous])
-        axis_type = (typeof(df[args[1]])<:PooledDataArray) ? :discrete : :continuous
+        axis_type = (typeof(dfn[args[1]])<:Union{CategoricalArray,NullableCategoricalArray}) ?
```

Comment:
I totally agree, that's done [here](https://github.com/JuliaPlots/StatPlots.jl/blob/master/src/groupederror.jl#L213): if there's an Array (not Categorical) of not numbers (e.g. strings or symbols) and the user didn't set `axis_type = :discrete` I give a warning and convert do discrete axis (if this is standard policy in StatPlots maybe I can remove the warning). Now the two checks (is array categorical and is `eltype<:Real`) got separated because I thought it's easier to check for the eltype after converting to normal arrays, but it's easier to check whether a column is categorical before converting.